src/main/java/com/example/springsample/SeatReservationService.java

```diff
@@ -0,0 +1,26 @@
+package com.example.springsample;
+
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Service
+public class SeatReservationService {
+
+    // This map stores the reservations with movieId as key and seatNumber as value
+    private final Map<String, Integer> reservations = new HashMap<>();
```

Comment:
**Suggestion:** The current implementation allows only one seat reservation per movie. To support multiple seat reservations for a single movie, consider changing the data structure of `reservations` to map a movieId to a list or set of reserved seats. [enhancement, importance: 9]
```suggestion
    private final Map<String, Set<Integer>> reservations = new HashMap<>();
```